apps/add_two_ints_service/src/main.c

```diff
@@ -10,55 +10,61 @@
 #include <std_msgs/msg/int32.h>
 #include "example_interfaces/srv/add_two_ints.h"
 
+rcl_init_options_t options;
+rcl_node_options_t node_ops;
+rcl_node_t node;
+rclc_support_t support;
+rcl_allocator_t allocator;
+rclc_executor_t executor;
+
+rcl_service_options_t service_options;
+rcl_service_t service;
+const rosidl_service_type_support_t * service_type_support;
+rcl_wait_set_t wait_set;
+rmw_request_id_t req_id;
```

Comment:
why you define it as global variables?

I guess when you use rclc_ functions, you dont need wait_set, service_options, node_ops any more. Have a look what other vars are not necessary any more.

Reply:
Reworked